Amplify/Categories/API/Operation/RetryableGraphQLOperation.swift

```diff
@@ -178,9 +195,35 @@ public final class RetryableGraphQLSubscriptionOperation<Payload: Decodable>: Op
     }
 
     public func shouldRetry(error: APIError?) -> Bool {
-        return attempts < maxRetries
+//        return attempts < maxRetries
+        
+        guard case let .operationError(errorDescription, recoverySuggestion, underlyingError) = error else {
+            return false
+        }
+        
+        if let authError = underlyingError as? AuthError {
+            switch authError {
+            case .signedOut, .notAuthorized:
+                return attempts < maxRetries
+            default:
+                return false
+            }
+        }
+        
+        // TODO: - How to distinguish errors?
+        // TODO: - Handle other errors
```

Comment:
looks like there is a whole list of errors to catch based on the strings https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore/src/sync/processors/subscription.ts#L692-L698